### PR TITLE
nettle-tests.py: Add Nettle crypto library tests

### PR DESCRIPTION
**Pull request by nasastry**
Add Nettle crypto library test cases

Signed-off-by: Nageswara R Sastry <rnsastry@linux.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
RHEL
```
JOB ID     : 2effbfe9713936359d7a44f3917a9a27783d6ba6
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-07-16T07.33-2effbfe/job.log
 (1/1) nettle-tests.py:Nettle.test: PASS (26.87 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-07-16T07.33-2effbfe/results.html
JOB TIME   : 27.52 s

JOB ID     : c73e6f7cb9fffd394d697a61dc316d23c407176f
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-07-16T07.33-c73e6f7/job.log
 (1/1) nettle-tests.py:Nettle.test: PASS (30.14 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-07-16T07.33-c73e6f7/results.html
JOB TIME   : 30.66 s

```

SLES:

```
JOB ID     : 98b53652df89b043be6253fbaa3733cc78fc72de
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-07-16T17.01-98b5365/job.log
 (1/1) nettle-tests.py:Nettle.test: PASS (28.77 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-07-16T17.01-98b5365/results.html
JOB TIME   : 29.02 s

```

File: security/nettle-tests.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    """
+
+    def setUp(self):
+        '''
+        Install the basic packages to support Nettle
+        '''
+
+        # Check for basic utilities
+        smm = SoftwareManager()
+        dd = distro.detect()
```
Comment (PraveenPenguin) [inline security/nettle-tests.py:37]: having variable name length has minmum  size of 3 char is always good to have , let me enhance travis to catch this

File: security/nettle-tests.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        deps = ['gcc', 'make', 'm4', 'libtool', 'automake', 'autoconf']
+        if dd.name in ['rhel', 'SuSE', 'fedora', 'centos', 'redhat']:
```
Comment (PraveenPenguin) [inline security/nettle-tests.py:39]: please remove `redhat` as distro check this deprecated

File: security/nettle-tests.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+        url = ("https://github.com/gnutls/nettle/archive/refs/heads/master.zip")
+        tarball = self.fetch_asset(url, expire='7d')
+        archive.extract(tarball, self.workdir)
+        self.sourcedir = os.path.join(self.workdir, 'nettle-master')
+        os.chdir(self.sourcedir)
+        process.run('autoreconf -ifv', ignore_status=True)
+        process.run('./configure', ignore_status=True)
+        if build.make(self.sourcedir) > 0:
```
Comment (PraveenPenguin) [inline security/nettle-tests.py:63]: if return is not `0` it can be failure case right ?

**Review by PraveenPenguin — CHANGES_REQUESTED** [review]

File: security/nettle-tests.py (hunk 1 above)
Comment (nasastry) [reply security/nettle-tests.py:37]: ok

File: security/nettle-tests.py (hunk 2 above)
Comment (nasastry) [reply security/nettle-tests.py:39]: ok

File: security/nettle-tests.py (hunk 3 above)
Comment (nasastry) [reply security/nettle-tests.py:63]: > if return is not `0` it can be failure case right ?

Yes `> 0` can be removed.

**Comment by nasastry** [comment]
After code change.

RHEL:
```
Fetching asset from nettle-tests.py:Nettle.test
JOB ID     : 237bf0859c8b78eb7f58f6aa91a438759e662999
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-07-26T11.13-237bf08/job.log
 (1/1) nettle-tests.py:Nettle.test: PASS (29.88 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-07-26T11.13-237bf08/results.html
JOB TIME   : 31.65 s
```

SLES:
```
Fetching asset from nettle-tests.py:Nettle.test
JOB ID     : 94af4e05ba213f054daa6d291dcc542ea608e17e
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-07-26T20.41-94af4e0/job.log
 (1/1) nettle-tests.py:Nettle.test: PASS (28.95 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-07-26T20.41-94af4e0/results.html
JOB TIME   : 30.49 s
```

**Review by PraveenPenguin — APPROVED** [review]
LGTM

**Pull request merged**